packages/v8env/src/headers.ts

```diff
@@ -10,175 +10,116 @@ import { logger } from "./logger"
  * @class
  * @param {Object} [init]
  */
-export default function Headers(init) {
-  Object.defineProperty(this, "_headerList", {
-    enumerable: false,
-    value: []
-  })
-  if (init) {
-    fill(this, init)
+export default class Headers {
```

Comment:
This could benefit from stronger typing by using `implements` like:

```typescript
export default class FlyHeaders implements Headers {
```

It can still implement more functions, but at least we'd know we support the spec. (assuming typescript's dom types are accurate)